thirdweb-dev/js
# [react, wallets] Paper wallet Native UI

## Problem solved

Instead of opening the Paper Modal, show the Custom UI in ConnectWallet component instead

## Changes made

- `PaperWallet.connect` method now takes `otp`, `recoveryCode` in args as well. when `otp` is provided - the wallet uses a headless login method from paper sdk

## How to test

- configured paperWallet in ThirdwebProvider, connect to it, expect to see a custom UI instead of the paper modal
- If the user has logged in from a new device - UI will also show an input to enter the recovery code
